src/lib/store.js

```diff
@@ -442,6 +472,11 @@ export default class Store {
 			purposes = [],
 		} = vendorList || {};
 
+		//collection of global and local vendor ids [global id, local id]
+		this.globalVendorIdsPresentOnList = new Map(vendors.filter(vendor => vendor.external_id).map(vendor => {
+			return [vendor.external_id, vendor.id];
+		}));
```

Comment:
I suggest `.map(vendor => [vendor.external_id, vendor.id])`. It will be more readable.